peer.go

```diff
@@ -116,11 +116,12 @@ func (l *PeerList) Add(hostPort string) *Peer {
 	return p
 }
 
-// Get returns a peer from the peer list, or nil if none can be found.
-func (l *PeerList) Get(prevSelected map[string]struct{}) (*Peer, error) {
+// GetNew returns a new, previously un-selected peer from the peer list, or nil,
+// if no new un-selected peer can be found.
```

Comment:
@prashantv So they'll have to do that awkward split error handling (like `EAGAIN` etc)

```go
peer, err := pl.GetNew(prev)
if err != nil {
  if err != ErrNoNewPeers {
    // log / wrap err
    return ...
  } else {
    // Do something different with the knowledge.
  }
}
// Use the peer without checking for nil.

```